src/sagas/favorite.js

```diff
@@ -0,0 +1,43 @@
+import { takeLatest, put, call } from 'redux-saga/effects'
+
+import { removeItem, setItem } from '../services/Storage'
+import { updateFavoritesArray, getFavsFromStorage } from '../utils'
+import {
+  FAVORITE_CLICK,
+  FAVORITE_FETCH_INITIAL_GIFS,
+  favoriteFetchInitialGifsSuccess,
+  favoriteFetchInitialGifsFailed
+} from '../actions'
+import giphy from '../services/GIPHY'
+
+export function * onFavoriteClick (action) {
+  try {
+    const favorites = getFavsFromStorage()
```

Comment:
You should use `call` here, `call(getFavsFromStorage)` to avoid calling directly this side effect here.